### PR TITLE
enhancements on command plugin

### PR DESCRIPTION
**Pull request by stbischof**
- should run without scr-impl

### CODE REVIEW

File: biz.aQute.gogo.commands.provider/play.bndrun [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 -runrequires: \
 	osgi.identity;filter:='(osgi.identity=biz.aQute.gogo.commands.provider)',\
-	osgi.identity;filter:='(osgi.identity=biz.aQute.shell.sshd.provider)'
+	osgi.identity;filter:='(osgi.identity=biz.aQute.shell.sshd.provider)',\
+	bnd.identity;id='org.apache.felix.log',\
+	bnd.identity;id='org.apache.felix.http.jetty',\
+	bnd.identity;id='org.apache.felix.http.servlet-api'
 -runfw: org.apache.felix.framework;version='[6.0.2,6.0.2]'
--runee: JavaSE-1.8
+-runee: JavaSE-11
```
Comment (pkriens) [inline biz.aQute.gogo.commands.provider/play.bndrun:9]: I cannot go beyond 1.8 ...

File: biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/DTOFramework.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+	@Descriptor(value = "deliveres the FrameworkDTO")
+	public FrameworkDTO frameworkDTO() {
+		final Bundle bundle = context.getBundle(0);
+		final FrameworkDTO frameworkDTO = bundle.adapt(FrameworkDTO.class);
+		return frameworkDTO;
+	}
+
+	@Descriptor(value = "deliveres the BundleDTOs")
```
Comment (pkriens) [inline biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/DTOFramework.java:43]: spelling error,  happens several times in this file.

File: biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/ExportPackageBundleTracker.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.Constants;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.util.tracker.BundleTracker;
+
+public class ExportPackageBundleTracker extends BundleTracker<Bundle> implements Closeable {
```
Comment (pkriens) [inline biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/ExportPackageBundleTracker.java:14]: I do not think this is worth the effort as we discussed. To do this properly you need to do a lot of work and you might still be wrong due to class loader constraints. Just loading a class from the package, protected by a try/catch is the only reliable way to find out you can import something. And it is a lot simpler.

File: cnf/central.mvn [hunk 4]
```diff
@@ -113,6 +113,7 @@ org.osgi:org.osgi.service.coordinator:1.0.2
 org.osgi:org.osgi.service.event:1.4.0
 org.osgi:org.osgi.service.http.whiteboard:1.1.0
 org.osgi:org.osgi.service.http:1.2.1
+org.osgi:org.osgi.service.jaxrs:jar:1.0.0
```
Comment (pkriens) [inline cnf/central.mvn:116]: I never expected to ever see Jax rs in a project I control :-) I hate it with a vengance :-)

Did you ever look at my OpenAPI project?

**Review by pkriens — CHANGES_REQUESTED** [review]

File: cnf/central.mvn (hunk 4 above)
Comment (stbischof) [reply cnf/central.mvn:116]: years ago when i created an openapi from the jaxrsruntimedto

File: biz.aQute.gogo.commands.provider/play.bndrun (hunk 1 above)
Comment (stbischof) [reply biz.aQute.gogo.commands.provider/play.bndrun]: was a mistake. 1.8 is good

File: biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/base/Activator.java [hunk 5]
```diff
@@ -35,15 +39,43 @@
 	@Override
 	public void start(BundleContext context) throws Exception {
 		this.context = context;
+		context.registerService(DTOFormatter.class, formatter, new Hashtable<>());
```
[the remark this answers is not in the document]
Comment (stbischof) [reply biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/base/Activator.java]: removed it. was an leftover

File: biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/ExportPackageBundleTracker.java (hunk 3 above)
Comment (stbischof) [reply biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/ExportPackageBundleTracker.java:11]: I changed it. Now the ExportPackageBundleTracker does no check than the the registration protected by the try/catch. But is tryes it again an new installed bundles.

I am still not happy with the exceptions. but do not see an other way.

File: biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/DTOFramework.java (hunk 2 above)
Comment (stbischof) [reply biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/DTOFramework.java]: thx done

**Pull request merged**